server/api/submission_handler.py

```diff
@@ -23,7 +18,20 @@ def create_submission(contest_id):
     data = jwt.decode(token, app.app.config['JWT_SECRET'], algorithms=['HS256'])
     current_user = User.query.filter_by(username=data['username']).first()
 
-    image = 'https://'+S3_BUCKET+'.s3.amazonaws.com/'+request.form["file_name"]
+    # Replace all the ":", ".", and " " with other charachters to have a consistent URL.
+    # AWS would replace these charachters with others such as "+" or "%" and the link in the database would not match the actual link on AWS.
+    file_name = (str(datetime.now()) + "_" + request.form["file_name"]).replace(":", "-").replace(".", "-").replace(" ", "_")
+    folder_path = 'contest_id_{}/user_id_{}/{}'.format(
+            contest_id,
+            current_user.id,
+            file_name)
+    image = 'https://'+S3_BUCKET+".s3."+S3_REGION+'.amazonaws.com/'+folder_path
+
+    try:
+        s3.upload_fileobj(request.files['file'], S3_BUCKET, folder_path)
```

Comment:
this is where we should be getting the success message from amazon. here is where you can probably find what amazon saved the image as